CMakeLists.txt

```diff
@@ -223,11 +223,25 @@ if (NOT CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE")
     endif ()
 endif()
 
+if (CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "MINSIZEREL")
+    set (STRIP_HEAVY_DEBUG_SYMBOLS_DEFAULT ON)
+else()
+    set (STRIP_HEAVY_DEBUG_SYMBOLS_DEFAULT OFF)
+endif()
+# Provides faster linking and lower binary size.
```

Comment:
About Release builds (l. 226): One would expect to build everything without debug symbols (-g0). But that's not the case right now.

```
cmake -DCMAKE_BUILD_TYPE=Release -S . -B build
vim build/build.ninja   # -g0 is enabled only for few files
```
I suppose the gains for building Release with -g0 will be bigger than with strip_heavy_debug_symbols.

Reply:
Yes, but who uses `CMAKE_BUILD_TYPE=Release`?

Reply:
I have seen "Release" used in closed-source software but you are right, in CH "Release" probably isn't used, not even for packaging. So we might keep things as is.